Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -24,135 +25,143 @@ export default class VirtualMachineScaleSet {
-                                    "commandToExecute": this.taskParameters.customScriptCommand
+                        var extensionMetadata: azureModel.VMExtensionMetadata = null;
+                        var customScriptExtension: azureModel.VMExtension = null;
+                        if(!!this.taskParameters.customScriptUrl) {
+                            extensionMetadata = this._getCustomScriptExtensionMetadata(osType);
+                            customScriptExtension = {
+                                name: "CustomScriptExtension" + Date.now().toString(),
+                                properties: {
+                                    type: extensionMetadata.type,
+                                    publisher: extensionMetadata.publisher,
+                                    typeHandlerVersion: extensionMetadata.typeHandlerVersion,
+                                    autoUpgradeMinorVersion: true,
+                                    settings: {
+                                        "fileUris": [ this.taskParameters.customScriptUrl ]
+                                    },
+                                    protectedSettings: {
+                                        "commandToExecute": this.taskParameters.customScriptCommand
+                                    }
                                 }
-                            }
-                        };
+                            };
 
-                        this._getExistingCustomScriptExtension(client, resourceGroupName, customScriptExtension, (error, matchingExtension, request, response) => {
+                            var matchingExtension = await this._getExistingCustomScriptExtension(client, resourceGroupName, customScriptExtension);
                             // if extension already exists, remove it
                             if(!!matchingExtension) {
-                                this._deleteAndInstallCustomScriptExtension(client, resourceGroupName, matchingExtension, customScriptExtension, resolve, reject, (error, matchingExtension, request, response) => {
-                                    this._updateImageInternal(client, resourceGroupName, resolve, reject);
-                                });
+                                await this._deleteAndInstallCustomScriptExtension(client, resourceGroupName, matchingExtension, customScriptExtension);
+                                await this._updateImageInternal(client, resourceGroupName);
                             } else {
-                                this._installCustomScripExtension(client, resourceGroupName, customScriptExtension, resolve, reject, (error, result, request, response) => {
-                                    this._updateImageInternal(client, resourceGroupName, resolve, reject);
-                                });
+                                await this._installCustomScripExtension(client, resourceGroupName, customScriptExtension);
```

Comment:
For my knowledge, is this the right way to update image (to do it in two steps). Would it lead to smooth upgrade; I thought earlier logic of merged update was better for upgrade; i.e. would it lead to zero downtime?